### PR TITLE
Convert store/storetest/job_store.go t.Fatal calls into assert/requir…

### LINKED ISSUE #12786: Migrate tests from "store/storetest/job_store.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `store/storetest/job_store.go` file.

The expected way to migrate this is to go to the `store/storetest/job_store.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19450

### PR DESCRIPTION
**Pull request by rv404674**
#### Summary.
Convert store/storetest/job_store.go t.Fatal calls into assert/require calls.

#### Ticket Link
Fixes #12786

### CODE REVIEW

File: store/storetest/job_store.go [hunk 1]
```diff
@@ -77,15 +74,11 @@ func testJobGetAllByType(t *testing.T, ss store.Store) {
...
-	} else if len(received) != 2 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[0].Id && received[1].Id != jobs[0].Id {
-		t.Fatal("should've received first jobs")
-	} else if received[0].Id != jobs[1].Id && received[1].Id != jobs[1].Id {
-		t.Fatal("should've received second jobs")
-	}
+	received, err := ss.Job().GetAllByType(jobType)
+	require.Nil(t, err)
+	require.Equal(t, 2, len(received), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: You can use directly `require.Len` which is more semantic here.

File: store/storetest/job_store.go [hunk 2]
```diff
@@ -120,23 +113,16 @@ func testJobGetAllByTypePage(t *testing.T, ss store.Store) {
...
-	} else if len(received) != 2 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[2].Id {
-		t.Fatal("should've received newest job first")
-	} else if received[1].Id != jobs[0].Id {
-		t.Fatal("should've received second newest job second")
-	}
+	received, err := ss.Job().GetAllByTypePage(jobType, 0, 2)
+	require.Nil(t, err)
+	require.Equal(t, 2, len(received), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: `require.Len` is a better option here

File: store/storetest/job_store.go [hunk 3]
```diff
@@ -120,23 +113,16 @@ func testJobGetAllByTypePage(t *testing.T, ss store.Store) {
...
-	if received, err := ss.Job().GetAllByTypePage(jobType, 2, 2); err != nil {
-		t.Fatal(err)
-	} else if len(received) != 1 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[1].Id {
-		t.Fatal("should've received oldest job last")
-	}
+	received2, err := ss.Job().GetAllByTypePage(jobType, 2, 2)
+	require.Nil(t, err)
+	require.Equal(t, 1, len(received2), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: `require.Len` is a better option here

File: store/storetest/job_store.go [hunk 4]
```diff
@@ -167,23 +153,16 @@ func testJobGetAllPage(t *testing.T, ss store.Store) {
...
-	} else if len(received) != 2 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[2].Id {
-		t.Fatal("should've received newest job first")
-	} else if received[1].Id != jobs[0].Id {
-		t.Fatal("should've received second newest job second")
-	}
+	received, err := ss.Job().GetAllPage(0, 2)
+	require.Nil(t, err)
+	require.Equal(t, 2, len(received), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: `require.Len` is a better option here

File: store/storetest/job_store.go [hunk 5]
```diff
@@ -167,23 +153,16 @@ func testJobGetAllPage(t *testing.T, ss store.Store) {
...
-	if received, err := ss.Job().GetAllPage(2, 2); err != nil {
-		t.Fatal(err)
-	} else if len(received) < 1 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[1].Id {
-		t.Fatal("should've received oldest job last")
-	}
+	received2, err := ss.Job().GetAllPage(2, 2)
+	require.Nil(t, err)
+	require.LessOrEqual(t, 1, len(received2), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: This is not the same test, I think here we should use `require.GreaterOrEqual`

File: store/storetest/job_store.go [hunk 6]
```diff
@@ -226,15 +205,11 @@ func testJobGetAllByStatus(t *testing.T, ss store.Store) {
...
-	} else if len(received) != 3 {
-		t.Fatal("received wrong number of jobs")
-	} else if received[0].Id != jobs[1].Id || received[1].Id != jobs[0].Id || received[2].Id != jobs[2].Id {
-		t.Fatal("should've received jobs ordered by CreateAt time")
-	} else if received[1].Data["test"] != "data" {
-		t.Fatal("should've received job data field back as saved")
-	}
+	received, err := ss.Job().GetAllByStatus(status)
+	require.Nil(t, err)
+	require.Equal(t, 3, len(received), "received wrong number of jobs")
```
Comment (jespino) [inline store/storetest/job_store.go]: `require.Len` is a better option here

File: store/storetest/job_store.go [hunk 7]
```diff
@@ -226,15 +205,11 @@ func testJobGetAllByStatus(t *testing.T, ss store.Store) {
...
+	require.False(t, received[0].Id != jobs[1].Id || received[1].Id != jobs[0].Id || received[2].Id != jobs[2].Id, "should've received jobs ordered by CreateAt time")
```
Comment (jespino) [inline store/storetest/job_store.go]: Here you can use three lines using different `require.Equal` for each or clause.

File: store/storetest/job_store.go [hunk 8]
```diff
@@ -360,24 +335,21 @@ func testJobUpdateOptimistically(t *testing.T, ss store.Store) {
 		"Foo": "Bar",
 	}
 
-	if updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS); err2 != nil {
-		if updated {
-			t.Fatal("should have failed due to incorrect old status")
-		}
+	updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS)
+	if updated {
```
Comment (jespino) [inline store/storetest/job_store.go]: This is not exactly the same, but I think it should be deterministic, so I think we can remove the `if updated` clause there, and only check for the `require.Nil(t, err)`

File: store/storetest/job_store.go [hunk 9]
```diff
@@ -360,24 +335,21 @@ func testJobUpdateOptimistically(t *testing.T, ss store.Store) {
...
-	require.True(t, updated)
+	require.True(t, updated2)
 
 	updatedJob, err := ss.Job().Get(job.Id)
 	require.Nil(t, err)
 
-	if updatedJob.Type != job.Type || updatedJob.CreateAt != job.CreateAt || updatedJob.Status != job.Status || updatedJob.LastActivityAt <= job.LastActivityAt || updatedJob.Progress != job.Progress || updatedJob.Data["Foo"] != job.Data["Foo"] {
-		t.Fatal("Some update property was not as expected")
-	}
+	require.False(t, updatedJob.Type != job.Type || updatedJob.CreateAt != job.CreateAt || updatedJob.Status != job.Status || updatedJob.LastActivityAt <= job.LastActivityAt || updatedJob.Progress != job.Progress || updatedJob.Data["Foo"] != job.Data["Foo"], "Some update property was not as expected")
```
Comment (jespino) [inline store/storetest/job_store.go]: This can be splitted in different `require.Equal` calls, one per each or in the `if`

**Comment by jasonblais** [comment]
Hi @rv404674, thanks for your contribution! Let us know if you have any questions about the feedback above, we'd be happy to help :)

**Comment by jasonblais** [comment]
Hey @rv404674, given you have two other PRs open with no recent activity, I'll close this one since this has more feedback to address. If you'd still like to work on all these, let us know! I'll just open it up for other interested community members to work on in the meantime :)

**Pull request closed without merging**